ykbh/src/lib.rs

```diff
@@ -0,0 +1,267 @@
+use std::alloc::{alloc, dealloc, Layout};
+use std::convert::{TryFrom, TryInto};
+use ykpack::{
+    self, Constant, ConstantInt, IPlace, Local, LocalDecl, Statement, Terminator, TypeId,
+    UnsignedInt,
+};
+use yktrace::sir::SIR;
+
+const SUSIZE: usize = std::mem::size_of::<usize>();
+
+pub struct SIRInterpreter {
+    map: *mut u8,
```

Comment:
Can this be `Box<[u8]>` or `*mut [u8]`?

Reply:
What are the advantages of this?

Reply:
The first one will avoid unsafe allocation and de-allocation code. The second one merely makes bound checking much easier.

Reply:
I think I prefer `*mut u8` over this, since we'll have to do a lot of pointer arithmetic. Though I could be convinced otherwise if there's a way to easily do that with `[u8]` as well.